nixos/modules/services/desktops/dunst.nix

```diff
@@ -0,0 +1,457 @@
+        description = ''
+        '';
+      };
+
+      iconPosition = mkOption {
+        type = types.str;
+        default = "off";
+        description = ''
+        '';
+      };
+
+      iconFolders = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = ''
+        '';
+      };
+
+    };
+
+
+    frame = {
+      width = mkOption {
+        type = types.int;
+        default = 3;
+        description = ''
+        '';
+      };
+
+      color = mkOption {
+        type = types.str;
+        default = ''#aaaaaa'';
+        description = ''
+        '';
+      };
+
+    };
+
+    shortcuts = {
+      close = mkOption {
+        type = types.str;
+        default = "ctrl+space";
+        description = ''
+        '';
+      };
+
+      closeAll = mkOption {
+        type = types.str;
+        default = "ctrl+shift+space";
+        description = ''
+        '';
+      };
+
+      history = mkOption {
+        type = types.str;
+        default = "ctrl+Escape";
+        description = ''
+        '';
+      };
+
+      context = mkOption {
+        type = types.str;
+        default = "ctrl+shift+period";
+        description = ''
+        '';
+      };
+
+    };
+
+    urgency = {
+      low = {
+        background = mkOption {
+          type = types.str;
+          default = ''#222222'';
+          description = ''
+          '';
+        };
+
+        foreground = mkOption {
+          type = types.str;
+          default = ''#888888'';
+          description = ''
+          '';
+        };
+
+        timeout = mkOption {
+          type = types.int;
+          default = 10;
+          description = ''
+          '';
+        };
+      };
+      normal = {
+        background = mkOption {
+          type = types.str;
+          default = ''#285577'';
+          description = ''
+          '';
+        };
+
+        foreground = mkOption {
+          type = types.str;
+          default = ''#ffffff'';
+          description = ''
+          '';
+        };
+
+        timeout = mkOption {
+          type = types.int;
+          default = 10;
+          description = ''
+          '';
+        };
+      };
+      critical = {
+        background = mkOption {
+          type = types.str;
+          default = ''#900000'';
+          description = ''
+          '';
+        };
+
+        foreground = mkOption {
+          type = types.str;
+          default = ''#ffffff'';
+          description = ''
+          '';
+        };
+
+        timeout = mkOption {
+          type = types.int;
+          default = 0;
+          description = ''
+          '';
+        };
+
+      };
+    };
+
+    extraConfig = mkOption {
+      type = types.string;
+      default = "";
+      description = "Extra configuration to append to the config file.";
+    };
+
+  };
+
+
+###### implementation
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ pkgs.dunst ];
+    systemd.user.services.dunst = {
```

Comment:
I'm curious, is this systemd service actually used for you? If you are in an X session with a running dunst, and run
```console
$ systemctl --user status dunst.service
```
does it show as active or inactive? I suspect it will show inactive since this is something I've been having difficulties with in Home Manager.